### ISSUE TITLE
Compile error when using "xc16plusplus_only.sh linux"

### ISSUE THREAD

**Issue by marcio-ao** [fabio-d/xc16plusplus#1]
I am trying to run "xc16plusplus_only.sh linux" on a Debian Stretch system, and it fails on compiler detection:

```
checking compiler i386-linux-gcc -m32 -O2 -pedantic -fomit-frame-pointer ... no
checking compiler i386-linux-gcc -O2 -pedantic -fomit-frame-pointer ... no
checking compiler gcc -m32 -O2 -pedantic -fomit-frame-pointer ... no
checking compiler gcc -O2 -pedantic -fomit-frame-pointer ... yes
checking compiler gcc -O2 -pedantic -fomit-frame-pointer has sizeof(long)==4... no
checking compiler i386-linux-icc -no-gcc ... no
checking compiler icc -no-gcc ... no
checking compiler i386-linux-cc -O ... no
checking whether cc is gcc... yes
checking compiler cc -m32 -O2 -pedantic -fomit-frame-pointer ... no
checking compiler cc -O2 -pedantic -fomit-frame-pointer ... yes
checking compiler cc -O2 -pedantic -fomit-frame-pointer has sizeof(long)==4... no
configure: error: could not find a working compiler, see config.log for details
```

I've included the config logs. Any ideas?

[config.log](https://github.com/fabio-d/xc16plusplus/files/1785743/config.log)
[config.log](https://github.com/fabio-d/xc16plusplus/files/1785745/config.log)

BTW, I work for [AlephObjects](https://www.alephobjects.com/) and we pride ourselves in using only openly licensed products. Thank you for your work in making a open alternative to the PIC compiler!

**Comment by fabio-d** [comment]
Hi! It fails to proceed because you are on a 64 bit system: the script wants to build a 32 bit compiler and expects to run a on 32-bit system (in other words, it is not designed to cross-compile the compiler).

There are three approaches you might want to follow:
1. Compile on a 32-bit debian system/virtual machine/chroot - I've just tested and it compiles fine on a Debian Stretch 32-bit VM(*); then copy the resulting executables to your 64-bit system;
2. Try to build a native 64-bit compiler executable;
3. Try to cross-compile a 32-bit compiler on your 64-bit system (you will need `g++-multilib libc6-dev-i386` and a lot of patience to cope with the build errors).

The first approach is what I do for the binaries I release: I compile on a fedora-22-i386 chroot inside a 64-bit system.

To be honest when i wrote the script I chose not to consider the second option at all, given that Microchip's other executables are 32-bit anyway.

The third option would be interesting, I tried to make it work in the past but I gave up because some configure commands seemed to ignore me and stubbornly built 64-bit libraries. If you manage to do it, I will be happy to accept your patch!

(*) = All I had to do was to replace `#!/bin/sh` with `#!/bin/bash` in xc16plusplus_only.sh and install packages `build-essential bison flex m4`.
